lib/irc/Scheduler.js

```diff
@@ -0,0 +1,64 @@
+/*eslint no-invalid-this: 0*/
+"use strict"
+var Promise = require("bluebird");
+var logging = require("../logging");
+var log = logging.get("scheduler");
+var Queue = require("../util/Queue.js");
+
+/**
+ * An IRC connection scheduler. Enables ConnectionInstance to reconnect
+ * in a way that queues reconnection requests and services the FIFO queue at a
+ * rate determined by ircServer.getReconnectIntervalMs().
+ */
+
+ var queues = {};
+
+function procFn (item) {
+    return Promise.delay(item.addedDelayMs).then(item.fn);
+}
+
+function getQueue (server) {
+    let q = queues[server.domain];
+
+    if (!q) {
+        q = new Queue(procFn, server.getReconnectIntervalMs());
+
+        queues[server.domain] = q;
+    }
+    return q;
+}
+
+var Scheduler = {
+    // Returns a promise that will be resolved when retryConnection returns a promise that
+    //  resolves, in other words, when the connection is made. The promise will reject if the
+    //  promise returned from retryConnection is rejected.
+    reschedule: Promise.coroutine(function*(server, addedDelayMs, retryConnection, nick) {
+        var q = getQueue(server);
+
+        var promise = q.enqueue(
+            `Scheduler.reschedule ${server.domain} ${nick}`,
+            {
+                fn: retryConnection,
+                addedDelayMs: addedDelayMs
+            }
+        );
+
+        log.info(
+            `Tryed to queue scheduled promise for ${server.domain} ${nick}` +
```

Comment:
![image](https://cloud.githubusercontent.com/assets/7190048/18083720/3252c28e-6e9c-11e6-82ce-9878ff82b707.png)

By this point you have already queued it, you aren't trying to do so.